### PR TITLE
doc: Add doc/release-notes/release-notes-25.0.md

### PR DESCRIPTION
**Pull request by maflcko**
Also, fix a typo in another doc.

### CODE REVIEW

File: doc/release-notes/release-notes-25.0.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Work in progress release notes are on the dev wiki:
```
Comment (fanquake) [inline doc/release-notes/release-notes-25.0.md:1]: Not the release notes?

**Review by stickies-v — APPROVED** [review]
ACK fab19a8ae30c7f2b9347f70f16799e14dc1970b1, confirmed that the release-notes are identical to what was reviewed in https://github.com/bitcoin/bitcoin/pull/27686/

Suggesting some minor improvements to the release notes, but perhaps this is not a suitable place?

**Comment by fanquake** [comment]
> Suggesting some minor improvements to the release notes, but perhaps this is not a suitable place?

They should have been suggested before the notes were finalised in #27686. This is just copying them into master.

File: doc/release-notes/release-notes-25.0.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+frequently tested on them.  It is not recommended to use Bitcoin Core on
+unsupported systems.
+
+Notable changes
+===============
+
+P2P and network changes
+-----------------------
+
+- Transactions of non-witness size 65 and above are now allowed by mempool
```
[the remark this answers is not in the document]
Comment (stickies-v) [reply doc/release-notes/release-notes-25.0.md:46]: Ah, good point. `bytes`, then, as per the PR title? The lack of unit is upsetting high school teachers worldwide.

**Review by fanquake — APPROVED** [review]
ACK 034cb5ad4d4b72cf1ba5b153a558fcf6a8afa9aa

**Pull request merged**